File: plugin-tester-java/src/main/java/example/myapp/statefulhelloworld/GreeterActor.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * license agreements; and to You under the Apache License, version 2.0:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * This file is part of the Apache Pekko project, derived from Akka.
 */

/*
 * Copyright (C) 2018-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package example.myapp.statefulhelloworld;

import org.apache.pekko.actor.AbstractActor;
import org.apache.pekko.actor.Props;

// #actor
public class GreeterActor extends AbstractActor {

  public static class ChangeGreeting {
    public final String newGreeting;
    public ChangeGreeting(String newGreeting) {
      this.newGreeting = newGreeting;
    }
  }
  public static class GetGreeting {}
  public static GetGreeting GET_GREETING = new GetGreeting();

  public static class Greeting {
    public final String greeting;
    public Greeting(String greeting) {
      this.greeting = greeting;
    }
  }

  public static Props props(final String initialGreeting) {
    return Props.create(GreeterActor.class, () -> new GreeterActor(initialGreeting));
  }

  private Greeting greeting;

  public GreeterActor(String initialGreeting) {
    greeting = new Greeting(initialGreeting);
  }

  public AbstractActor.Receive createReceive() {
    return receiveBuilder()
        .match(GetGreeting.class, this::onGetGreeting)
        .match(ChangeGreeting.class, this::onChangeGreeting)
        .build();
  }

  private void onGetGreeting(GetGreeting get) {
    getSender().tell(greeting, getSelf());
  }

  private void onChangeGreeting(ChangeGreeting change) {
    greeting = new Greeting(change.newGreeting);
  }
}
// #actor

File: plugin-tester-scala/src/main/scala/example/myapp/typedhelloworld/GreeterActor.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * license agreements; and to You under the Apache License, version 2.0:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * This file is part of the Apache Pekko project, which was derived from Akka.
 */

/*
 * Copyright (C) 2018-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package example.myapp.typedhelloworld

import org.apache.pekko
import pekko.actor.typed.scaladsl.Behaviors
import pekko.actor.typed.{ ActorRef, Behavior }

// #actor
object GreeterActor {
  sealed trait GreetingCommand
  case class ChangeGreeting(newGreeting: String) extends GreetingCommand
  case class GetGreeting(replyTo: ActorRef[Greeting]) extends GreetingCommand

  case object GetGreeting
  case class Greeting(greeting: String)

  def apply(initialGreeting: String): Behavior[GreetingCommand] = (new GreeterActor(initialGreeting)).createBehavior()
}

class GreeterActor(initialGreeting: String) {
  import GreeterActor._

  var greeting = Greeting(initialGreeting)

  def createBehavior(): Behavior[GreetingCommand] = Behaviors.receiveMessage {
    case ChangeGreeting(newGreeting) =>
      greeting = Greeting(newGreeting)
      createBehavior()
    case GetGreeting(replyTo) =>
      replyTo ! greeting
      Behaviors.same
  }
}
// #actor
